assets/app/view/game/buy_sell_shares.rb

```diff
@@ -214,18 +215,22 @@ def render_short
       # Allow privates or minors to be exchanged for shares if they have the ability
       def render_exchanges
         children = []
-        source_entities = @game.companies + @game.minors
+        # 1858 can have the same entity appear as both a private company and a
+        # minor company. Use uniq to get rid of any potential duplicates.
+        source_entities = (@game.companies + @game.minors).uniq
```

Comment:
can we simplify this by having a subpar ui?

Reply:
This was put in as the same entities appear in both the `@game.companies` and the `@game.minors` arrays. These entities also have exchange abilities, and this was leading to the exchange buttons being duplicated.

If this `uniq` call is a problem then it should be possible to work around this by moving the entities between the companies and minors arrays as needed, instead of putting them in both. It will make the 1858 game code a little more complicated, but shouldn't be too difficult.

Reply:
This has been removed, it was no longer needed after the removal of the hybrid company/minor objects.